### PR TITLE
Fix #1632 #1635 #1644 Continuous logging added to compass

### LINKED ISSUE #1632: Add functionality to continuously record data in compass
**Actual Behaviour**
Currently record function in compass stores instantaneous value 

**Expected Behaviour**
Compass should record data until stop button is pressed , like other instruments

**LogCat for the issue**
N/A

**Screenshots of the issue**
N/A

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1632 #1635  #1644 

**Changes**: 
I have made following changes

- **Inherited** compass activity from **PSLabSensor activity** for consistency throughout the code
- Added functionality to **continuously log and play** data in compass instrment
- The logged data of compass was **not visible in Logged Data activity** previously. I have fixed it
- in the CSV generated for the log of compass, **selected axis and Location** is added, which was previuosly missing.

**Screenshot/s for the changes**: 
Screenshot_20190413-180724
Screenshot_20190413-180700
Screenshot_20190413-180655

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**:
[compass.apk.zip](https://github.com/fossasia/pslab-android/files/3076192/compass.apk.zip)

### CODE REVIEW

**Comment by neel1998** [comment]
@mariobehling @CloudyPadmal @harsh-2711 @cweitat @abhinavraj23  kindly review

**Comment by neel1998** [comment]
@cweitat  please review

**Comment by ho-dor** [comment]
@neel1998 please see the conflicting files. Thanks

**Comment by neel1998** [comment]
@CloudyPadmal @abhinavraj23  kindly review this

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
         android:roundIcon="@drawable/app_icon_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
+        <activity android:name=".activity.NewCompassActivity"></activity>
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml]: Why is this renamed to "new"? Can't we have the same class with changes?

Comment (neel1998) [reply app/src/main/AndroidManifest.xml]: I tried using the same name. There were some unexpected issues with the PSLab Sensor model. That's why changed the name. There some issues with ubinder. It can't find the 'compass' view when the activity name is kept the same. I could not figure out the reason. Can you look once by yourself

Comment (adityastic) [reply app/src/main/AndroidManifest.xml]: @neel1998 Im sure changing the name never give unexpected issues.

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+    private float currentDegree = 0f;
+    public Boolean writeHeaderToFile = true;
+    private int direction;
+    private SensorManager sensorManager;
+    private Sensor sensor;
+    private Timer graphTimer;
+    private long startTime, block;
+    private ArrayList<CompassData> recordedCompassArray;
+    CompassData compassData = new CompassData();
+    NewCompassActivity compassActivity;
```
Comment (adityastic) [inline app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: both can be private

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+                            /* Pass for another refresh round */
+                        }
+                    }
+                });
+            }
+        }, 0, updatePeriod);
+    }
+
+    private void visualizeData() {
+        timeElapsed = ((System.currentTimeMillis() - startTime) / updatePeriod);
```
Comment (adityastic) [inline app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: you can declare it here itself

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (CloudyPadmal) [reply app/src/main/AndroidManifest.xml]: Probably there were caches and files were dirty. Have you tried a clean rebuild? :thinking:

Comment (neel1998) [reply app/src/main/AndroidManifest.xml]: I rebuilt. It is working fine now. Sorry for my mistake.

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java (hunk 2 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: i will do that

**Comment by neel1998** [comment]
@CloudyPadmal  I have made the necessary changes. Kindly review

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -21,6 +21,7 @@
         android:roundIcon="@drawable/app_icon_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
+        <activity android:name=".activity.CompassActivity"></activity>
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml]: Change 
`<activity android:name=".activity.CompassActivity"></activity>`
to
`<activity android:name=".activity.CompassActivity"/>`

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java [hunk 5]
```diff
@@ -62,6 +62,10 @@ protected void onCreate(Bundle savedInstanceState) {
                 getSupportActionBar().setTitle(caller);
                 categoryData = LocalDataLog.with().getTypeOfSensorBlocks(getString(R.string.multimeter));
                 break;
+            case "Compass":
+                getSupportActionBar().setTitle(caller);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: What do you think @neel1998 , since this line is the same for all these cases, would it throw any error or malfunction if we put `getSupportActionBar().setTitle(caller);` before the switch statement?

File: app/src/main/java/io/pslab/models/CompassData.java [hunk 6]
```diff
@@ -36,4 +47,50 @@ public String getBz() {
...
+    }
+
+    public double getLon() {
+        return lon;
+    }
+
+    @Override
+    public String toString() {
+        return "Block - " + block + ", Time - " + time + ", Compass_X - " + Bx + ", Compass_Y - " + By + ", Compass_Z - " + Bz + ", Compass_axis" + Axis + ", Lat - " + lat + ", Lon - " + lon;
+
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/models/CompassData.java]: remove

File: app/src/main/res/layout/activity_compass.xml [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:fitsSystemWindows="true">
 
     <ScrollView
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:layout_marginTop="?attr/actionBarSize">
+        >
```
Comment (CloudyPadmal) [inline app/src/main/res/layout/activity_compass.xml]: Close the bracket at line 10

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1106,5 +1109,5 @@
     <string name="no_data_fetched">No Data Fetched</string>
     <string name="recorder">Recorder</string>
     <string name="save_graph">Save Graph</string>
-
+    <string name="value_null">0</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Don't add at the end of file

File: app/src/main/java/io/pslab/fragment/CompassDataFragment.java [hunk 9]
```diff
@@ -0,0 +1,559 @@
...
+            zAxisMagneticField.setText(String.valueOf(degree));
+        }
+
+        @Override
+        public void onAccuracyChanged(Sensor sensor, int accuracy) {
+
+        }
+    };
+
+    private void setCompassAnimation(float degree) {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/CompassDataFragment.java]: What does this animation do?

**Comment by CloudyPadmal** [comment]
Also squash the commits into one. Probably rebasing would help here.

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java (hunk 5 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: I think that can be done. I am on it

File: app/src/main/java/io/pslab/models/CompassData.java (hunk 6 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/models/CompassData.java]: You mean the extra line at the end?

Comment (CloudyPadmal) [reply app/src/main/java/io/pslab/models/CompassData.java]: Line 94

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java (hunk 5 above)
Comment (CloudyPadmal) [reply app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: I mean, first check if it's breaking anything. If not you can update it here itself :+1:

Comment (neel1998) [reply app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: The changes are working fine

**Comment by neel1998** [comment]
@CloudyPadmal can you please guide me how can I squash non consecutive commits using rebase

**Comment by CloudyPadmal** [comment]
@neel1998 https://github.com/fossasia/pslab-android#how-to-git-squash will help with it

**Comment by neel1998** [comment]
@CloudyPadmal  I have made necessary changes and have also squashed the commits. Kindly review and if valid then merge. Thanks

**Comment by neel1998** [comment]
@CloudyPadmal  kindly have a look at this, there are some feature needs to be implemented for the compass which can to be done without merging this PR. Thanks

**Comment by CloudyPadmal** [comment]
We can merge this once the conflict is resolved

**Comment by neel1998** [comment]
@CloudyPadmal  i have resolved the conflicts . Kindly review. I have also added the save_graph menu option in the PSLabSensor class. Kindly check the attached apk.
[compass.zip](https://github.com/fossasia/pslab-android/files/3169786/compass.zip)

**Comment by neel1998** [comment]
@CloudyPadmal  kindly review this

**Comment by neel1998** [comment]
@CloudyPadmal  kindly review and merge this as soon as possible so that I can start working g on gyroscope PR conflicts
